pkg/internal/api/validation/registry.go

```diff
@@ -35,8 +35,8 @@ type Registry struct {
 	validateUpdateRegister map[schema.GroupVersionKind]ValidateUpdateFunc
 }
 
-type ValidateFunc func(req *admissionv1.AdmissionRequest, obj runtime.Object) field.ErrorList
-type ValidateUpdateFunc func(req *admissionv1.AdmissionRequest, oldObj, obj runtime.Object) field.ErrorList
+type ValidateFunc func(req *admissionv1.AdmissionRequest, obj runtime.Object) (field.ErrorList, []string)
+type ValidateUpdateFunc func(req *admissionv1.AdmissionRequest, oldObj, obj runtime.Object) (field.ErrorList, []string)
```

Comment:
Further to that, we have `field.ErrorList` which has the notion of a path, helping to point people at where exactly an error is.

It seems a shame to _not_ have that kind of rich information for warnings. Maybe even something like this could work:

```go
type WarningList field.ErrorList
```

Reply:
Thanks- I agree that splitting makes sense!
Would it makes sense if it was:
1. Removes legacy functions (so that it's more straightforward to write e2e tests for 2)
2. Use cert-manager/crypto instead of meyskens/crypto + deprecates the keyAlgorithm- this would allow us to fork upstream golang/crypto & cherry-pick alternative cert chains (current verson from meyskens has an older version of golang/crypto with our custom functionality for ACME EAB) and once we use upstream for ACME EAB we have to get rid of keyAlgorithm at least partially.
3. Implements warnings

Reply:
That sounds spot on :+1: